lisa/environment.py

```diff
@@ -330,18 +331,20 @@ def create_node_from_requirement(
 
         return node
 
-    def get_information(self) -> Dict[str, str]:
-        final_information: Dict[str, str] = {}
+    def get_information(self, force_run: bool = True) -> Dict[str, str]:
+        if self._information_cache and not force_run:
```

Comment:
check like below, so if dict updated, but nothing got, it doesn't run again.

```python
        if self._information_cache is None and not force_run:
```

Reply:
Yes, that makes sense. Updated.

Reply:
Should I change the function return type to None|Dict[str,str]?

Some of the existing usages of this function expects Dict[str,str]

Reply:
No, the dict can keep the same returned type.